src/shared/json.c

```diff
@@ -253,9 +253,7 @@ static JsonVariant *json_variant_formalize(JsonVariant *v) {
                 return json_variant_unsigned(v) == 0 ? JSON_VARIANT_MAGIC_ZERO_UNSIGNED : v;
 
         case JSON_VARIANT_REAL:
-                DISABLE_WARNING_FLOAT_EQUAL;
-                return json_variant_real(v) == 0.0 ? JSON_VARIANT_MAGIC_ZERO_REAL : v;
-                REENABLE_WARNING;
+                return iszero_relaxed(json_variant_real(v)) ? JSON_VARIANT_MAGIC_ZERO_REAL : v;
```

Comment:
No, this change is wrong. The fact that a floating-point number has subnormal representation is an internal detail of the implementation of the fp standard. It is exposed because it has effect on precision, but it is not something that we care about.

The warning is stupid, and we need to keep disabling it.